main.go

```diff
@@ -111,6 +112,30 @@ func handleSignals(fCache *filecache.FileCache, mList *memberlist.Memberlist) {
 	os.Exit(130)                // Ctrl-C received or equivalent
 }
 
+// configureNewRelic sets up and starts a Gorelic agent if we have a
+// New Relic license available.
+func configureNewRelic() *gorelic.Agent {
+	nrLicense := os.Getenv("NEW_RELIC_LICENSE_KEY")
+	var agent *gorelic.Agent
+	if nrLicense == "" {
+		log.Info("No New Relic license found, not starting an agent")
+
+		return nil
```

Comment:
Won't the accesses to `agent.Tracer` freak out if the agent is nil?

Reply:
I worked around that in the Http code, but looks like the Redis code doesn't work well